### DISCUSSION TITLE
"Connection setup timed out in state" errors and reporting

### DISCUSSION THREAD

**Discussion by konalegi** [edenhill/librdkafka#4352] [Q&A]
We from time to time receive the following errors 
> sasl_ssl://host-name-removed: Connection setup timed out in state CONNECT (after 30064ms in state CONNECT)

Like a couple of times per day, while investigating I have found the following [description](https://github.com/confluentinc/librdkafka/blob/e75de5be191b6b8e9602efc969f4af64071550de/src/rdkafka_conf.c#L565C29-L570)

> "Maximum time allowed for broker connection setup "
     "(TCP connection setup as well SSL and SASL handshake). "
     "If the connection to the broker is not fully functional after this "
     "the connection will be closed and retried.", 

It says that the connection will be retried, how safe it's to ignore such kind of messages? Are the errors pop up when all retries are exhausted, or does it throw on each retry?

Thanks!

**Comment by prathmeshzepto** [comment]
Heyy @konalegi facing same error, Did you get any explanation on this ?

**Reply by konalegi** [reply]
As far as I remember, I have simply ignored that type of errors and haven't seen any effect on metrics, although I'm still not sure if I done it right.
